app/views/helpers/BtaPartialHelper.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.helpers
+
+import models.{ObligationModel, Overdue, Open}
+import play.api.i18n.Messages
+import play.twirl.api.Html
+import utils.ImplicitDateFormatter._
+import utils.ImplicitCurrencyFormatter._
+
+object BtaPartialHelper {
+  
+  def whichStatus(model: ObligationModel)(implicit messages: Messages) = model.getObligationStatus match {
+    case open: Open =>
+      Html(
+        s"""
+           |<p id="report-due">${messages("bta_partial.next_due", model.due.toLongDate)}</p>
+           |<a id="obligations-link" href=${controllers.routes.ObligationsController.getObligations().url}>${messages("bta_partial.deadlines_link")}</a>
+         """.stripMargin.trim
+      )
+    case overdue: Overdue.type =>
+      Html(
+        s"""
+           |<p id="report-due">${messages("bta_partial.next_due", model.due.toLongDate)}</p>
+           |<a id="obligations-link" href=${controllers.routes.ObligationsController.getObligations().url}>${messages("bta_partial.deadlines_link")}</a>
+         """.stripMargin.trim
+      )
+    case _ =>
```

Comment:
I assume a sealed trait is returned from getObligationStatus (if it's not then it should be). If it is, then you don't need a `case _ =>` as the compiler will know that the pattern match is exhaustive. So this can be removed :smile: